### PR TITLE
feat: 100% documented public members

### PR DESCRIPTION
**Pull request by josxha**
- Adds doc strings to all missing public members
- With this pr `public_member_api_docs` will get active. This enforces future PRs to keep the 100% documentation score of public members. Private members won't be affected.

### CODE REVIEW

File: lib/src/gestures/interactive_flag.dart [hunk 1]
```diff
@@ -25,6 +25,7 @@ abstract class InteractiveFlag {
       scrollWheelZoom |
       rotate;
 
+  /// Use this flag if you want to disable all gesture interactions.
```
Comment (JaffaKetchup) [inline lib/src/gestures/interactive_flag.dart:28]: Avoid 'you', and avoid explaining WHY this should be used in the first sentence of a doc. Instead explain WHAT is does.

https://dart.dev/effective-dart/documentation#prefer-starting-function-or-method-comments-with-third-person-verbs

```suggestion
  /// Disable all gesture interactions
```

File: lib/src/layer/attribution_layer/rich/source.dart [hunk 2]
```diff
@@ -9,6 +9,9 @@ import 'package:meta/meta.dart';
 sealed class SourceAttribution extends StatelessWidget {
   const SourceAttribution._({super.key, this.onTap});
 
+  /// This callback gets called if the user clicks or taps on the attribution
+  /// source. Most of the time you would want to open a website with the terms
+  /// of service of the tile provider.
```
Comment (JaffaKetchup) [inline lib/src/layer/attribution_layer/rich/source.dart:14]:
```suggestion
  /// This callback gets called when the user clicks or taps on the
  /// attribution source.
  /// 
  /// Most tile providers will require you to link to their terms of
  /// service, which can be done through this callback.
```

File: lib/src/layer/polygon_layer/polygon.dart [hunk 3]
```diff
@@ -26,6 +26,9 @@ class Polygon<R extends Object> {
   /// The color of the [Polygon] outline.
   final Color borderColor;
 
+  /// Weather holes on the [Polygon] should have a border too or not.
```
Comment (JaffaKetchup) [inline lib/src/layer/polygon_layer/polygon.dart:29]:
```suggestion
  /// Whether holes should have borders
```

File: lib/src/layer/tile_layer/tile_coordinates.dart [hunk 4]
```diff
@@ -2,6 +2,8 @@ import 'dart:math';
 
 import 'package:meta/meta.dart';
 
+/// The tile coordinates are based on maths' [Point] class but extended with
+/// the zoom level.
```
Comment (JaffaKetchup) [inline lib/src/layer/tile_layer/tile_coordinates.dart:6]: Consider a brief explanation of how tile coordinates work, or a link to the documentation site: https://docs.fleaflet.dev/getting-started/explanation#slippy-map-convention?

File: lib/src/layer/tile_layer/tile_image_view.dart [hunk 5]
```diff
@@ -3,6 +3,7 @@ import 'dart:collection';
 import 'package:flutter_map/flutter_map.dart';
 import 'package:flutter_map/src/layer/tile_layer/tile_range.dart';
 
+/// A
```
Comment (JaffaKetchup) [inline lib/src/layer/tile_layer/tile_image_view.dart:6]: Needs finishing.

**Review by JaffaKetchup — CHANGES_REQUESTED** [review]
Mostly LGTM, just a few things to think about...

File: lib/src/layer/tile_layer/tile_image_view.dart (hunk 5 above)
Comment (josxha) [reply lib/src/layer/tile_layer/tile_image_view.dart]: woops, that's an oversight. Thanks for bringing this up 😄 fixed

File: lib/src/layer/tile_layer/tile_coordinates.dart (hunk 4 above)
Comment (josxha) [reply lib/src/layer/tile_layer/tile_coordinates.dart:14]: Added more informations. Feel free to improve or enhance. Not sure how to describe best so I added the link to the docs as well.

**Review by JaffaKetchup — APPROVED** [review]
LGTM! Huge thanks, a lot of work here.

**Pull request merged**